File: Lua/001.lua
function sum_multiples(limit)
    sum = 0
    for i=0,limit-1 do
        if i % 3 == 0 or i % 5 == 0 then
            sum = sum + i
        end
    end
    return sum
end

print(sum_multiples(10))
print(sum_multiples(1000))

--[[
Intertesting alternate implementation
]]--
function tri(n)
    return n * (n + 1) / 2
end

function sum35(n)
    n = n - 1
    return (
        3 * tri(math.floor(n / 3)) +
        5 * tri(math.floor(n / 5)) -
        15 * tri(math.floor(n / 15))
    )
end

print(sum35(1000))
print(sum35(1e+20))

File: Go/001.go
package main

import "fmt"

func main() {
	fmt.Println(sum_multiples(10, []int{3, 5}))   // 23
	fmt.Println(sum_multiples(1000, []int{3, 5})) // 233168
}

func sum_multiples(limit int, multiples []int) int {
	// fmt.Println(limit)
	// fmt.Println(multiples)
	sum := 0
	for i := 1; i < limit; i++ {
		// fmt.Println(i)
		for _, multiple := range multiples {
			// fmt.Println(multiple)
			if i%multiple == 0 {
				sum += i
				break
			}
		}
	}

	return sum
}
